torch/testing/_internal/common_methods_invocations.py

```diff
@@ -4309,16 +4309,96 @@ def sample_inputs_narrow_copy(op_info, device, dtype, requires_grad, **kwargs):
         tensor = make_tensor(shape, dtype=dtype, device=device, low=None, high=None,
                              requires_grad=requires_grad)
         yield SampleInput(tensor, args=args)
+        # narrow also accepts the start argument being a Tensor
+        if is_narrow:
+            yield SampleInput(tensor, args=(args[0], torch.tensor(args[1]), args[2]))
```

Comment:
Thanks to @peterbell10, we can now write sample inputs "naturally" without specifying the `args` parameter:

```
yield SampleInput (tensors, args[0], ...)
```

This PR doesn't have to do this, btw, but ideally the `shape_and_args` tuple wouldn't pack the args into their own tuple. It would make the code more readable not to have to refer to `args[0]`. I understand it was probably done that way because of the historic requirement that the `args` parameter to the `SampleInput` constructor required a sequence.